weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/utilities.ts

```diff
@@ -286,6 +286,39 @@ export const objectVersionNiceString = (ov: ObjectVersionSchema) => {
   return result;
 };
 
+export function parseObjectVersionUrlPathToRefUri(
+  urlPath: string | undefined
+): string | null {
+  if (!urlPath) {
+    return null;
+  }
+
+  // Remove leading and trailing slashes
+  const cleanPath = urlPath.replace(/^\/|\/$/g, '');
+  const segments = cleanPath.split('/');
+
+  // We need at least 6 segments to be a valid object version path
+  if (segments.length < 6) {
+    return null;
+  }
+  const entity = segments[0];
+  const project = segments[1];
+  const objectId = segments[3];
+  const versionHash = segments[5];
```

Comment:
it feels a bit suspicious that this is the first place we are doing a Url to object key conversion. I understand why, but it feels like the object page router would be doing this as well. That being said I can't find a useful existing function to do this....

Reply:
yeah like Tim said: `usePeekLocation`